pkg/chunk/chunk_store.go

```diff
@@ -490,17 +490,24 @@ func (c *baseStore) lookupIdsByMetricNameMatcher(ctx context.Context, from, thro
 	} else if err != nil {
 		return nil, err
 	}
-	level.Debug(log).Log("matcher", matcher, "entries", len(entries))
+	level.Debug(log).Log("matcher", formatMatcher(matcher), "entries", len(entries))
 
 	ids, err := c.parseIndexEntries(ctx, entries, matcher)
 	if err != nil {
 		return nil, err
 	}
-	level.Debug(log).Log("matcher", matcher, "ids", len(ids))
+	level.Debug(log).Log("matcher", formatMatcher(matcher), "ids", len(ids))
 
 	return ids, nil
 }
 
+func formatMatcher(matcher *labels.Matcher) string {
```

Comment:
This function avoids logging nil matcher, which works, but indirectly via panic and recover. That confuses debugger attached from Goland, which wants to breakpoint on each panic.